docs/site/tutorials/protocol_oriented_generics.ipynb

```diff
@@ -0,0 +1,1141 @@
+{
```

Comment:
<p>Perhaps this tutorial can introduce protocols without starting with shortcomings in alternative approaches? Given that OOP may mean different things for different people and different languages, this may invite controversy which I think can be avoided. An alternative to the current wording would be skipping this section altogether, explaining how protocols work on compelling examples and then discussing pros and cons in comparison with designs that don't involve protocols.</p>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/swift/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>